web/src/settings_org.js

```diff
@@ -887,6 +887,37 @@ export function init_dropdown_widgets() {
     });
     settings_components.set_can_access_all_users_group_widget(can_access_all_users_group_widget);
     can_access_all_users_group_widget.setup();
+
+    const can_create_public_channel_group_widget = new dropdown_widget.DropdownWidget({
+        widget_name: "realm_can_create_public_channel_group",
+        get_options: () =>
+            user_groups.get_realm_user_groups_for_dropdown_list_widget(
+                "can_create_public_channel_group",
+                "realm",
+            ),
+        $events_container: $("#settings_overlay_container #organization-permissions"),
+        item_click_callback(event, dropdown) {
+            dropdown.hide();
+            event.preventDefault();
+            event.stopPropagation();
+            settings_components.can_create_public_channel_group_widget.render();
+            settings_components.save_discard_realm_settings_widget_status_handler(
+                $("#org-stream-permissions"),
+            );
+        },
+        tippy_props: {
+            placement: "bottom-start",
+        },
+        default_id: realm.realm_can_create_public_channel_group,
+        unique_id_type: dropdown_widget.DataTypes.NUMBER,
+        on_mount_callback(dropdown) {
+            $(dropdown.popper).css("min-width", "300px");
+        },
+    });
+    settings_components.set_can_create_public_channel_group_widget(
+        can_create_public_channel_group_widget,
+    );
+    can_create_public_channel_group_widget.setup();
```

Comment:
This is probably a big enough chunk of code that we will want to extract it as a reusable function taking just a few parameters.

Reply:
Yes, can do this as a follow-up.

Reply:
Added this to my TODO list, so resolving this conversation.